README.md

```diff
@@ -2,80 +2,203 @@
 [![Coverage Status](https://coveralls.io/repos/broadinstitute/cromwell/badge.svg?branch=develop)](https://coveralls.io/r/broadinstitute/cromwell?branch=develop)
 
 
-Cromwell
-========
+# Cromwell
 
-Workflow engine using [WDL](SPEC.md) as the workflow and task language.
+Workflow engine using [WDL](https://github.com/broadinstitute/wdl/blob/wdl2/SPEC.md) as the workflow and task language.
 
-API Documenation
-----------------
+<!---toc start-->
+
+* [Cromwell](#cromwell)
+  * [API Documenation](#api-documenation)
+  * [Scala API Usage](#scala-api-usage)
+  * [REST API](#rest-api)
+    * [POST /workflows](#post-workflows)
+    * [GET /workflow/:id/status](#get-workflowidstatus)
+    * [GET /workflow/:id/outputs](#get-workflowidoutputs)
+* [Developer](#developer)
+  * [Generate WDL Parser](#generate-wdl-parser)
+  * [Generating and Hosting ScalaDoc](#generating-and-hosting-scaladoc)
+  * [Architecture](#architecture)
+    * [cromwell.parser](#cromwellparser)
+    * [cromwell.binding](#cromwellbinding)
+    * [cromwell.backend](#cromwellbackend)
+    * [cromwell.engine](#cromwellengine)
+
+<!---toc end-->
+
+## API Documenation
 
 API Documentation can be found [here](http://broadinstitute.github.io/cromwell/scaladoc).
 
-Generate WDL Parser
--------------------
+Note that this may not be completely up to date or even useful at this time.
 
-Install the latest version of [Hermes](http://github.com/scottfrazer/hermes), then run the following command within this directory:
+## Scala API Usage
 
+The main entry point into the parser is the `WdlNamespace` object.  A WDL file is considered a namespace, and other namespaces can be included by using the `import` statement (but only with an `as` clause).
+
+```scala
+import cromwell.binding.WdlNamespace
+
+object Main extends App {
+  val ns = WdlNamespace.load(new File("test.wdl"))
+  val ns2 = WdlNamespace.load("workflow wf {}")
+  val workflows = ns.workflows
+  val tasks = ns.tasks
+  val commands = ns.tasks.map{_.command}
+}
 ```
-hermes generate src/main/resources/grammar.hgr --language=java --directory=src/main/java --name=wdl --java-package=cromwell.parser --java-use-apache-commons
+
+To access only the parser, use the `AstTools` library, as follows:
+
+```scala
+import cromwell.binding.WdlNamespace
+
+object Main extends App {
+  val ast = AstTools.getAst(new File(args(0)))
+
+  // Second parameter is a descriptor about where the first string came from.
+  // Most of the time this would be the URI of where the text was loaded from,
+  // but there are no restrictions on what the string can be.
+  val astFromString = AstTools.getAst("workflow wf {}", "string")
```

Comment:
Hmm yeah, it is.  I'm okay with that for now, especially considering this will very likely not be used by any third-parties in the near future.